### PR TITLE
Java: Fix linting issues

### PR DESCRIPTION
**Pull request by johnoliver**
### Contribution Checklist

- [ ] The code builds clean without any errors or warnings
- [ ] The PR follows the [SK Contribution Guidelines](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md) and the [pre-submission formatting script](https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md#development-scripts) raises no violations
- [ ] All unit tests pass, and I have added new tests where possible
- [ ] I didn't break anyone :smile:

### CODE REVIEW

File: java/aiservices/openai/src/main/java/com/microsoft/semantickernel/aiservices/openai/chatcompletion/OpenAIChatCompletion.java [hunk 1]
```diff
@@ -575,6 +608,12 @@ public static class Builder extends ChatCompletionService.Builder {
 
         @Override
         public OpenAIChatCompletion build() {
+            if (client == null) {
```
Comment (dsgrieve) [inline java/aiservices/openai/src/main/java/com/microsoft/semantickernel/aiservices/openai/chatcompletion/OpenAIChatCompletion.java:611]: This differs from OpenAITextGenerationService Builder build() method. I think it's correct there and not here. Our SK API should always throw an SKException type (which may wrap a cause).

File: java/samples/pom.xml [hunk 2]
```diff
@@ -126,6 +126,10 @@
                             <target>${maven.compiler.release}</target>
                             <encoding>${project.build.sourceEncoding}</encoding>
                             <showWarnings>true</showWarnings>
+                            <compilerArgs>
```
Comment (dsgrieve) [inline java/samples/pom.xml:129]: ? Is there something missing here?

File: java/samples/sample-code/src/main/java/com/microsoft/semantickernel/samples/syntaxexamples/Example06_TemplateLanguage.java [hunk 3]
```diff
@@ -105,6 +105,6 @@ Is it weekend time (weekend/not weekend)?
         // Show the result
         System.out.println("--- Prompt Function result");
         var result = kernel.invokeAsync(kindOfDay, null, String.class).block();
-        System.out.println(result.getResultVariable());
+        System.out.println(result.getResultVariable().getValue());
```
Comment (dsgrieve) [inline java/samples/sample-code/src/main/java/com/microsoft/semantickernel/samples/syntaxexamples/Example06_TemplateLanguage.java:108]: I thought there was a "getValue()" on FunctionResult that is a shorthand for this.
`<edit>`I see you do this in Example09. Best to be consistent. It doesn't matter to me which way.`</edit>`

File: java/samples/sample-code/src/main/java/com/microsoft/semantickernel/samples/syntaxexamples/Example09_FunctionTypes.java [hunk 4]
```diff
@@ -164,42 +174,103 @@ public String toString() {
...
             .invokeAsync(plugin.get("noInputComplexReturnType"),
                 null,
                 Temporal.class)
             .block();
+        System.out.println(result.getResult());
 
-        kernel
+        // Possibilities for return type combinations:
+        // | Method return type | Declared Function Return Type | Invocation Return type |
+        // |--------------------|-------------------------------|------------------------|
```
Comment (dsgrieve) [inline java/samples/sample-code/src/main/java/com/microsoft/semantickernel/samples/syntaxexamples/Example09_FunctionTypes.java:215]: Oh, sure. Make me use a monospaced typeface. ;^)

File: java/samples/sample-code/src/main/java/com/microsoft/semantickernel/samples/syntaxexamples/Example09_FunctionTypes.java [hunk 5]
```diff
@@ -164,42 +174,103 @@ public String toString() {
...
+        result = kernel
+            .invokeAsync(plugin.get("conversionScenarioA"),
+                null,
+                Instant.class)
+            .block();
+        System.out.println(result.getResult());
+
+        // Potentially unsupported scenario as it requires a downcast
+        /*
+        x = kernel
```
Comment (dsgrieve) [inline java/samples/sample-code/src/main/java/com/microsoft/semantickernel/samples/syntaxexamples/Example09_FunctionTypes.java:239]: What about a downcast method in the type converter? `<U extends T> Function<T, U> downcast`. Default implementation would just throw a SKException variant of a ClassCastException. There are some things that are easy to downcast, Number to Integer or Collection to List for example, but it obviously isn't an inline cast.

File: java/samples/sample-code/src/main/java/com/microsoft/semantickernel/samples/syntaxexamples/Example09_FunctionTypes.java [hunk 6]
```diff
@@ -377,18 +461,18 @@ public Mono<Void> InputStringTaskWithVoidResult(
...
             );
         }
 
         @DefineKernelFunction(
             name = "noInputComplexReturnType",
             returnType = "java.time.temporal.Temporal"
         )
-        public LocalDateTime noInputComplexReturnType() {
-            return LocalDateTime.now();
+        public OffsetDateTime noInputComplexReturnType() {
```
Comment (dsgrieve) [inline java/samples/sample-code/src/main/java/com/microsoft/semantickernel/samples/syntaxexamples/Example09_FunctionTypes.java:474]: Why did you choose OffsetDateTime here as opposed to ZonedDateTime?

File: java/semantickernel-api/src/main/java/com/microsoft/semantickernel/orchestration/contextvariables/ContextVariable.java [hunk 7]
```diff
@@ -1,23 +1,147 @@
...
+        if (t == null) {
+            return new ContextVariable<>(
+                ContextVariableTypes.getDefaultVariableTypeForClass(type),
+                null);
+        }
+
+        return of(t);
+    }
+
+    public static <T> ContextVariable<T> untypedOf(
```
Comment (dsgrieve) [inline java/semantickernel-api/src/main/java/com/microsoft/semantickernel/orchestration/contextvariables/ContextVariable.java:31]: Should this return `ContextVariable<?>`

File: java/semantickernel-api/src/main/java/com/microsoft/semantickernel/orchestration/contextvariables/ContextVariable.java [hunk 8]
```diff
@@ -1,23 +1,147 @@
...
+                        requestedResultType
+                    ),
+                    requestedResultType.cast(it)
+                );
+            }
+        }
+
+        ContextVariableType<T> requestedResultTypeVariable = null;
+
+        try {
```
Comment (dsgrieve) [inline java/semantickernel-api/src/main/java/com/microsoft/semantickernel/orchestration/contextvariables/ContextVariable.java:83]: suggest breaking these out into smaller functions so you could then use streams findFirst

**Review by dsgrieve — CHANGES_REQUESTED** [review]

File: java/semantickernel-api/src/main/java/com/microsoft/semantickernel/orchestration/contextvariables/ContextVariable.java (hunk 8 above)
Comment (johnoliver) [reply java/semantickernel-api/src/main/java/com/microsoft/semantickernel/orchestration/contextvariables/ContextVariable.java:83]: Unfortunately due to the multiple exit points this is hard to extract

File: java/samples/sample-code/src/main/java/com/microsoft/semantickernel/samples/syntaxexamples/Example09_FunctionTypes.java (hunk 6 above)
Comment (johnoliver) [reply java/samples/sample-code/src/main/java/com/microsoft/semantickernel/samples/syntaxexamples/Example09_FunctionTypes.java:474]: This is purely as our DateTimeContextVariable works on OffsetDateTime, if we add a converter for Local or Zoned date time we could use that here

File: java/samples/sample-code/src/main/java/com/microsoft/semantickernel/samples/syntaxexamples/Example09_FunctionTypes.java (hunk 4 above)
Comment (dsgrieve) [reply java/samples/sample-code/src/main/java/com/microsoft/semantickernel/samples/syntaxexamples/Example09_FunctionTypes.java:215]: I use ComicSans. 

JK

**Pull request merged**